### DISCUSSION TITLE
installing neolink on raspberry pi -error

### DISCUSSION THREAD

**Discussion by Pete54321** [thirtythreeforty/neolink#297] [General]
I think i have followed all the instructions carefully ... but

When I run:

   peter@raspberrypi:/usr/local/bin $ ./neolink --config my_config.toml

I get:

   Illegal instruction

Any ideas anyone?  I have a Reolink Lumus camera i am trying to access.

Thanks in advance

Pete

**Comment by QuantumEntangledAndy** [comment]
Most likely you are using the wrong binary for raspberry pis. Which one did you download.

**Reply by Pete54321** [reply]
Hi Andy

Its a PI2b ...  PiOS Desktop (32bit) Release 4.2 - tried the Debian armhf (Raspberry Pi 32-bit) ([download](https://nightly.link/thirtythreeforty/neolink/workflows/build/master/release-armhf-buster.zip)) link...

Also tried the PiOS 32 bit lite with the same effect.

Tried the Ubuntu/Debian x86 ([download](https://nightly.link/thirtythreeforty/neolink/workflows/build/master/release-i386-buster.zip))  Version - different error ...   -bash: /usr/local/bin/neolink: cannot execute binary file: Exec format error

Im running the command:

    neolink rtsp --config my_config.toml

Is there another binary to try?

Thanks for your support on this!

Pete

**Reply by QuantumEntangledAndy** [reply]
Ah pi2B. We compile for the armv7 processor with hardware floats (armhf). I think perhaps the cpu on the Pi2 lacks the necessary capabilities. Your best bet would be to compile it yourself.
